### ISSUE TITLE
Microsoft.NETCore.App', version '3.1.0' - no longer supported. Azure Microsoft Agents fail to run task

### ISSUE THREAD

**Issue by SA-Egress** [DBojsen/Azure-Data-Factory-CI-CD-tools#3]
Hi, 

.NET Core 3.1 Runtime which the task is dependent on is no longer supported by Microsoft. Microsofts self-hosted agents have begun to remove this leading to task failures.

https://dotnet.microsoft.com/en-us/download/dotnet/3.1/runtime?cid=getdotnetcore&os=windows&arch=x64

```
`You must install or update .NET to run this application.

App: D:\a\_tasks\DBojsenDataFactoryDeployment_e15848e4-4d49-4214-ba6d-7ec15f56197e\0.9.10\ConsoleApp\DBojsen.DataFactory.Deployment.exe
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '3.1.0' (x64)
.NET location: C:\Program Files\dotnet\

The following frameworks were found:
  6.0.5 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  6.0.18 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  6.0.19 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  7.0.7 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

Learn about framework resolution:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=3.1.0&arch=x64&rid=win10-x64
##[debug]Caught exception from task script.
##[debug]Error record:
##[debug]D:\a\_tasks\DBojsenDataFactoryDeployment_e15848e4-4d49-4214-ba6d-7ec15f56197e\0.9.10\deploy.ps1 : Errors occurred while deploying
##[debug]At line:1 char:1
##[debug]+ . 'D:\a\_tasks\DBojsenDataFactoryDeployment_e15848e4-4d49-4214-ba6d-7 ...
##[debug]+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
##[debug]    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
##[debug]    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,deploy.ps1
##[debug] 
##[debug]Script stack trace:
##[debug]Microsoft.PowerShell.Commands.WriteErrorException: Errors occurred while deploying
##[error]Errors occurred while deploying
##[debug]Processed: ##vso[task.logissue type=error]Errors occurred while deploying
##[debug]Processed: ##vso[task.complete result=Failed]`
```

**Comment by SuChuDom** [comment]
Facing the same issue.

**Comment by DBojsen** [comment]
Currently on vacation - will update asap

**Comment by DBojsen** [comment]
Hi @SA-accesso & @SuChuDom 

I have just released version 0.9.11, which is now using .Net 6 instead. 

Please check if this solves your issue. 

BR David

**Comment by SA-Egress** [comment]
Thanks @DBojsen. Issue resolved.

While we have your attention. Any ETA on when this will be out of preview? Is there still some work pending?
